data-service/src/main/java/uk/gov/gchq/palisade/service/data/request/AuditMessage.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.request;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.exception.PalisadeRuntimeException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+
+/**
+ * This is the parent class for Audit information.  It represents the common component of the data that is to be
+ * sent to audit service.
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AuditMessage {
+
+    protected static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @JsonProperty("userId")
+    protected final String userId; //Unique identifier for the user.
+
+    @JsonProperty("resourceId")
+    protected final String resourceId;  //Resource that that is being asked to access.
+
+    @JsonProperty("context")
+    protected final JsonNode context;   //Relevant context information about the request.
+
+    @Value("${spring.application.name}")
```

Comment:
This pulls the service name from the application yaml, which will always be data-service, it can be removed or left in. Junit tests that dont spin up a service will use the initialised variable, but services will use the spring.application.name (data-service)